app/web/features/profile/edit/EditProfile.tsx

```diff
@@ -101,6 +109,36 @@ export default function EditProfileForm() {
   const { regions, regionsLookup } = useRegions();
   const { languages, languagesLookup } = useLanguages();
 
+  const charsRemainingAboutMe = () => {
+    const watchAboutMe = watch("aboutMe");
+
+    if (
+      !watchAboutMe ||
+      watchAboutMe.length > completeProfileMinimumCharachters
+    ) {
+      return null;
+    } else {
+      const missingChars =
+        completeProfileMinimumCharachters - watchAboutMe.length;
+      return (
+        <Alert severity="warning">
+          <Trans i18nKey="profile:edit_profile_incomplete_helper.helper_text">
+            Please write{" "}
+            <strong>
+              {{
+                chars: t("profile:edit_profile_incomplete_helper.chars", {
+                  count: missingChars,
+                }),
+              }}
+            </strong>{" "}
+            in this section to complete your profile. You also need to upload a
+            profile picture.
+          </Trans>
+        </Alert>
+      );
+    }
+  };
```

Comment:
The way this is written feels like it could be its own component to me, and I'd consider using [`useWatch`](https://www.react-hook-form.com/api/usewatch/) in that separate component instead as according to the docs, it helps to isolate re-renders only to this bit of the component instead of this entire `EditProfile` component every time the `aboutMe` field updates

Reply:
But otherwise, I think what you've done here is fine and no need for Controllers!